preprocessors/charts/charts_utils.py

```diff
@@ -0,0 +1,54 @@
+# For a given point at a index, the function returns the number of
+# contiguous points(on the left side of the point) lower than the point value
+def getLowerPointsOnLeft(index, point, series_data):
+    count = 0
+    if index == 0:
+        return count
+    for data_point in series_data[index-1::-1]:
```

Comment:
So we're running these for loops each time for any chart data we get? What's the general purpose application of these outputs?

Reply:
We are running these loops to get the additional metrics associated the given chart. Once we have these metrics(lowerPointsOnLeft, higherPointsOnLeft, etc...) this information can be further used to calculate local maxima/local minima, global maxima or perform sliding window technique to remove noise from the chart